### PR TITLE
[Elastic-Agent] Do not use unversioned home with watcher binary path

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

The problem is when agent tries to invoke watcher which checks upgraded agent for failures and at the same time unversioned home path is set.

Usually agent took its home path which is `{top_path}/data/elastic-agent-{hash}/elastic-agent` and invoked `watch` sub-command on binary at that path.

When unversioned path is set it tries to run watcher from `{top_path}/data/elastic-agent` and as so it results in binary not found. 

Fixes: #25371

## Why is it important?

Reduce noise created by failng attempt to run this. This is just a case when running `container` sub-command which tempers paths. 

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

cc @michel-laterman

### CODE REVIEW

**Comment by michalpristas** [comment]
@blakerouse should i do this or should we move binary/create symlink in the setPath func

File: x-pack/elastic-agent/pkg/agent/application/upgrade/service.go [hunk 1]
```diff
@@ -200,8 +200,8 @@ func (p *noopPidProvider) Name() string { return "noop" }
 
 func (p *noopPidProvider) PID(ctx context.Context) (int, error) { return 0, nil }
 
-func invokeCmd() *exec.Cmd {
-	homeExePath := filepath.Join(paths.Home(), agentName)
+func invokeCmd(topPath string) *exec.Cmd {
```
Comment (ruflin) [inline x-pack/elastic-agent/pkg/agent/application/upgrade/service.go:203]: I saw the name topPath in the past? What does it exactly mean?

**Review by ruflin — COMMENTED** [review]
@michalpristas Can you reference the issue this is coming from?

**Comment by ruflin** [comment]
As a follow up, we should start to think on how we can test these things in an automated way.

File: x-pack/elastic-agent/pkg/agent/application/upgrade/service.go (hunk 1 above)
Comment (michalpristas) [reply x-pack/elastic-agent/pkg/agent/application/upgrade/service.go:203]: with the structure we have topPath is the top of agent directory structure e.g. `/etc/Elastic/Agent`
we also use a homePath which points to a home of a certain agent version e.g `/etc/Elastic/Agent/data/elastic-agent-abc123/`

**Comment by blakerouse** [comment]
The only time unversioned path is used is in the container. Self-upgrade is not supported in the container anyway, so could we just disable the watcher completely instead?

**Review by blakerouse — APPROVED** [review]
This change does seem okay and should still go in.

But related to my previous comment, when will the watcher run in a container?

**Comment by michalpristas** [comment]
@blakerouse it will not, the only case when it's running is when specifically told to using DEV flag [allowUpgrade part]

**Review by ruflin — APPROVED** [review]

**Pull request merged**